app/models/grant_submission/response.rb

```diff
@@ -45,11 +47,11 @@ class Response < ApplicationRecord
                             allow_nil: true
 
     validate :validate_by_response_type
-    validate :response_if_mandatory, if: -> { question.is_mandatory? && submission.submitted? }
-    validate :attachment_is_valid,   if: -> { document.attached? && submission.submitted? }
+    validate :response_if_mandatory, if: -> { question.is_mandatory? && (submission.submitted? || submission&.user_submitted_state == 'submitted') }
+    validate :attachment_is_valid,   if: -> { document.attached? }
```

Comment:
won't this throw an error if someone hasn't uploaded a document on a draft?

Reply:
The validation confirms that the question.response_type is file_upload to prevent files from being mistakenly attached to, say, a short_text question.